From 0203e838801fcdb400517e141d07b068620c7590 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?V=C3=ADctor=20Manuel=20J=C3=A1quez=20Leal?=
 <vjaquez@igalia.com>
Date: Wed, 19 Apr 2023 12:46:10 +0200
Subject: [PATCH 114/127] video: add GST_VIDEO_DMA_DRM_CAPS_MAKE

It is requried this macro because it doesn't define format field, neither
drm-format which is suppossed to be generated in run-time.
---
 girs/GstVideo-1.0.gir                              |  6 ++++++
 .../gst-libs/gst/video/video-info-dma.h            | 14 ++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/girs/GstVideo-1.0.gir b/girs/GstVideo-1.0.gir
index d20a9642d8..8cf98f3cdd 100644
--- a/girs/GstVideo-1.0.gir
+++ b/girs/GstVideo-1.0.gir
@@ -2812,6 +2812,12 @@ return to upstream/caller (indicating either GST_FLOW_ERROR or GST_FLOW_OK).</do
         </parameter>
       </parameters>
     </function-macro>
+    <constant name="VIDEO_DMA_DRM_CAPS_MAKE" value="video/x-raw(" c:type="GST_VIDEO_DMA_DRM_CAPS_MAKE" version="1.24">
+      <doc xml:space="preserve" filename="../subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h">Generic caps string for video wit DMABuf feature, for use in pad templates.
+As drm-format is supposed to be defined at run-time it's not predefined here.</doc>
+      <source-position filename="../subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h"/>
+      <type name="utf8" c:type="gchar*"/>
+    </constant>
     <function-macro name="VIDEO_ENCODER" c:identifier="GST_VIDEO_ENCODER" introspectable="0">
       <source-position filename="../subprojects/gst-plugins-base/gst-libs/gst/video/gstvideoencoder.h"/>
       <parameters>
diff --git a/subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h b/subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h
index 61f8319eca..aa4890ad5e 100644
--- a/subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h
+++ b/subprojects/gst-plugins-base/gst-libs/gst/video/video-info-dma.h
@@ -25,6 +25,20 @@
 
 G_BEGIN_DECLS
 
+/**
+ * GST_VIDEO_DMA_DRM_CAPS_MAKE:
+ *
+ * Generic caps string for video wit DMABuf feature, for use in pad templates.
+ * As drm-format is supposed to be defined at run-time it's not predefined here.
+ *
+ * Since: 1.24
+ */
+#define GST_VIDEO_DMA_DRM_CAPS_MAKE                                     \
+    "video/x-raw(" GST_CAPS_FEATURE_MEMORY_DMABUF "), "                 \
+    "width = " GST_VIDEO_SIZE_RANGE ", "                                \
+    "height = " GST_VIDEO_SIZE_RANGE ", "                               \
+    "framerate = " GST_VIDEO_FPS_RANGE
+
 typedef struct _GstVideoInfoDmaDrm GstVideoInfoDmaDrm;
 
 /**
-- 
2.25.1

